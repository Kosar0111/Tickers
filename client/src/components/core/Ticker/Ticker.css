.ticker {
  display: flex;
  align-items: center;
  justify-content: space-around;
  border: 1px solid grey;
  margin: 20px;
  padding: 20px;
  border-radius: 10px;
  width: 60vw;
  height: 5vh;
}
.label {
  display: block;
  margin-bottom: 10px;
  font-weight: bold;
}
.company {
  padding: 5px;
}
.img-down {
  width: 15px;
  height: 20px;
}
.change_percent {
  min-width: 100px;
  border-radius: 10px;
  padding: 5px;
  background-color: green;
}
.change_percent-negative {
  min-width: 100px;
  border-radius: 10px;
  padding: 5px;
  background-color: red;
}
.dividend,
.tickeexchanger,
.last_trade_time,
.price {
  padding: 5px;
}
.value {
  display: flex;
  justify-content: space-around;
}
