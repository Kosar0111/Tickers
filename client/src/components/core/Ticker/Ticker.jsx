import { useRef, useEffect } from 'react'
import down from '../../../assets/img/down.svg'
import up from '../../../assets/img/up.svg'
import './Ticker.css'

export const Ticker = (el) => {
  const { change, change_percent, dividend, exchange, last_trade_time, price, ticker } = el
  const prevEl = useRef(el)
  useEffect(() => {
    if (prevEl.current.price !== el.price) return (prevEl.current = el)
  }, [el])
  return (
    <div className="ticker">
      <div className="company">
        <span className="label">Ticker</span>
        {ticker}
      </div>
      <div
        className={prevEl.current.price < el.price ? 'change_percent' : 'change_percent-negative'}
      >
        <span className="label">Price</span>
        <div className="value">
          {prevEl.current.price > el.price ? (
            <img src={down} alt="down" className="img-down" />
          ) : (
            <img src={up} alt="down" className="img-down" />
          )}
          {price}
        </div>
      </div>
      <div
        className={prevEl.current.price < el.price ? 'change_percent' : 'change_percent-negative'}
      >
        <span className={'label'}>Change</span>
        <div className="value">
          {prevEl.current.price > el.price ? (
            <img src={down} alt="down" className="img-down" />
          ) : (
            <img src={up} alt="down" className="img-down" />
          )}
          {change}
        </div>
      </div>
      <div
        className={prevEl.current.price < el.price ? 'change_percent' : 'change_percent-negative'}
      >
        <span className="label">Change %</span>
        <div className="value">
          {prevEl.current.price > el.price ? (
            <img src={down} alt="down" className="img-down" />
          ) : (
            <img src={up} alt="down" className="img-down" />
          )}
          {change_percent}
        </div>
      </div>
      <div className="dividend">
        <span className="label">Divident</span>
        {dividend}
      </div>
      <div className="tickeexchanger">
        <span className="label">Exchange</span>
        {exchange}
      </div>
      <div className="last_trade_time">
        <span className="label">Trade time</span>
        {last_trade_time.slice(11, -5)}
      </div>
    </div>
  )
}
