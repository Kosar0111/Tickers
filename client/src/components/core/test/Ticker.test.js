import { render } from '@testing-library/react'
import { Ticker } from '../Ticker/Ticker'

const ticker = {
  ticker: 'AAPL',
  exchange: 'NASDAQ',
  price: 279.29,
  change: 64.52,
  change_percent: 0.84,
  dividend: 0.56,
  yield: 1.34,
  last_trade_time: '2021-04-30T11:53:21.000Z'
}
describe('Ticker', () => {
  it('should create item', () => {
    const component = render(<Ticker {...ticker} />)
    expect(component).toMatchSnapshot()
  })
})
