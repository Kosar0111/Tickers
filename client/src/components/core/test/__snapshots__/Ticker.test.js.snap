// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker should create item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ticker"
      >
        <div
          class="company"
        >
          <span
            class="label"
          >
            Ticker
          </span>
          AAPL
        </div>
        <div
          class="change_percent-negative"
        >
          <span
            class="label"
          >
            Price
          </span>
          <div
            class="value"
          >
            <img
              alt="down"
              class="img-down"
              src="up.svg"
            />
            279.29
          </div>
        </div>
        <div
          class="change_percent-negative"
        >
          <span
            class="label"
          >
            Change
          </span>
          <div
            class="value"
          >
            <img
              alt="down"
              class="img-down"
              src="up.svg"
            />
            64.52
          </div>
        </div>
        <div
          class="change_percent-negative"
        >
          <span
            class="label"
          >
            Change %
          </span>
          <div
            class="value"
          >
            <img
              alt="down"
              class="img-down"
              src="up.svg"
            />
            0.84
          </div>
        </div>
        <div
          class="dividend"
        >
          <span
            class="label"
          >
            Divident
          </span>
          0.56
        </div>
        <div
          class="tickeexchanger"
        >
          <span
            class="label"
          >
            Exchange
          </span>
          NASDAQ
        </div>
        <div
          class="last_trade_time"
        >
          <span
            class="label"
          >
            Trade time
          </span>
          11:53:21
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ticker"
    >
      <div
        class="company"
      >
        <span
          class="label"
        >
          Ticker
        </span>
        AAPL
      </div>
      <div
        class="change_percent-negative"
      >
        <span
          class="label"
        >
          Price
        </span>
        <div
          class="value"
        >
          <img
            alt="down"
            class="img-down"
            src="up.svg"
          />
          279.29
        </div>
      </div>
      <div
        class="change_percent-negative"
      >
        <span
          class="label"
        >
          Change
        </span>
        <div
          class="value"
        >
          <img
            alt="down"
            class="img-down"
            src="up.svg"
          />
          64.52
        </div>
      </div>
      <div
        class="change_percent-negative"
      >
        <span
          class="label"
        >
          Change %
        </span>
        <div
          class="value"
        >
          <img
            alt="down"
            class="img-down"
            src="up.svg"
          />
          0.84
        </div>
      </div>
      <div
        class="dividend"
      >
        <span
          class="label"
        >
          Divident
        </span>
        0.56
      </div>
      <div
        class="tickeexchanger"
      >
        <span
          class="label"
        >
          Exchange
        </span>
        NASDAQ
      </div>
      <div
        class="last_trade_time"
      >
        <span
          class="label"
        >
          Trade time
        </span>
        11:53:21
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
